bin/logstash.lib.sh

```diff
@@ -1,5 +1,29 @@
 unset CDPATH
-LOGSTASH_HOME=$(cd `dirname $0`/..; pwd)
+# This unwieldy bit of scripting is to try to catch instances where Logstash
+# was launched from a symlink, rather than a full path to the Logstash binary
+if [ -L $0 ]; then
```

Comment:
Plus, this code is for _finding_ the correct directory, so finding/including the source of a library in order to find the full path to the file is a full loop of pain :frowning: